src/StateNode.ts

```diff
@@ -105,6 +105,7 @@ class StateNode<
   TStateSchema extends StateSchema = any,
   TEvent extends OmniEventObject<EventObject> = OmniEventObject<EventObject>
 > {
+  public isXStateNode: true = true;
```

Comment:
Why not relying on `instanceof`? Mainly to be safe when using frames, `instanceof` is not reliable in such scenarios.

As to the name - totally bikesheddable.

Reply:
What about using `Symbol()`?

Reply:
Symbol wouldnt work, we could use Symbol.for though. IMHO it wouldn't really change things much and it wouldn't be friendly for older environments where Symbol (and its global cross frame registry) is not available

Reply:
I'm wondering too if we can just do `source instanceof StateNode` - that would suffice.

Reply:
> Why not relying on instanceof? Mainly to be safe when using frames, instanceof is not reliable in such scenarios.

This is a bummer, but the scenario is possible - maybe it's not something popular to do, but there are valid use cases for this and `instanceof` wouldn't cover it.

Consider a parent page with src-less iframe holding a script tag which in turn creates a machine, but later that machine is being interpreted by a parent page (which is possible as you can synchronously access `window.parent` from such frame without being blocked).

This issue is often visible when manipulating the DOM and trying to use checks like `foo instanceof Element` which fails if an element was created in the other frame (`frame.Element !== window.Element`).

So even if it's not super popular we can just not worry about such things by using a simple property with a boolean value (something which i.e. React does too - a component's instance has `isReactComponent` property, https://overreacted.io/how-does-react-tell-a-class-from-a-function/ )

Reply:
@davidkpiano any more thoughts on this? 😃